lib/http.js

```diff
@@ -125,7 +125,41 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
   // update keystore config
   app.post(
     `${routes.keystore}`,
-    _authorizeZcapInvocation(),
+    _authorizeZcapInvocation({
+      getExpectedTarget: async ({req}) => {
+        const {body: {id: configId}} = req;
+        const requestUrl = `${req.protocol}://${req.get('host')}${req.url}`;
+        if(configId !== requestUrl) {
+          throw new BedrockError(
+            'The request URL does not match the configuration ID.',
+            'URLMismatchError', {
+              public: true,
+              configId,
+              requestUrl,
+            });
+        }
+        return {expectedTarget: configId};
+      },
+      onError: ({error}) => {
+        // cause must be a public BedrockError to be surfaced to the HTTP client
+        let cause;
+        if(error instanceof BedrockError) {
+          cause = error;
+        } else {
+          cause = new BedrockError(
+            error.message,
+            error.name || 'PermissionDenied', {
```

Comment:
Is `PermissionDenied` still a thing? I thought we moved over to `NotAllowedError`.